api/src/main/java/edu/cornell/mannlib/vitro/webapp/edit/n3editing/configuration/preprocessors/LimitRemovalsToLanguage.java

```diff
@@ -0,0 +1,94 @@
+package edu.cornell.mannlib.vitro.webapp.edit.n3editing.configuration.preprocessors;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.jena.rdf.model.Literal;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.rdf.model.StmtIterator;
+
+import edu.cornell.mannlib.vitro.webapp.rdfservice.filter.LanguageFilteringUtils;
+
+/**
+ * A model change preprocessor that discards triples with language tags
+ * in the retractionsModel that do not match the specified language, unless
+ * the additionsModel also contains a new value for the same subject and 
+ * predicate in that language, or no values in any language are added for the 
+ * same subject and predicate (such as when an entire resource is deleted) .
+ */
+public class LimitRemovalsToLanguage implements ModelChangePreprocessor {
+
+    private static final Log log = LogFactory.getLog(LimitRemovalsToLanguage.class);
+    private String language;
+    
+    /**
+     * @param locale the Java locale object representing the language
+     * to which edits should be limited.  May not be null.
+     */
+    public LimitRemovalsToLanguage(Locale locale) {
+        if(locale == null) {
+            throw new IllegalArgumentException("Locale may not be null.");    
+        }            
+        this.language = LanguageFilteringUtils.localeToLanguage(locale);
+    }
+    
+    /**
+     * @param language string representing the RDF language tag to which
+     * edits should be limited. May not be null.
+     */
+    public LimitRemovalsToLanguage(String language) {
+        if(language == null) {
+            throw new IllegalArgumentException("Language may not be null.");    
+        }
+        this.language = language;
+    }
+    
+    @Override
+    public void preprocess(Model retractionsModel, Model additionsModel,
+            HttpServletRequest request) {                 
+        log.debug("limiting changes to " + language);
+        List<Statement> eliminatedRetractions = new ArrayList<Statement>();
+        StmtIterator sit = retractionsModel.listStatements();
+        while(sit.hasNext()) {
+            Statement stmt = sit.next();
+            if(!stmt.getObject().isLiteral()) {
+                continue;
```

Comment:
I am always a little wary of `continue` and `break` in loops, as they make following the logic slightly trickier.
In this particular case, the `continue` can easily be removed with:
```
if (stmt.getObject().isLiteral()) {
    ...
```

If, however, you have a reason that I am not seeing for keeping the `continue`, I defer to your discretion.

Reply:
Yep, it was pretty pointless in this case.  Pushed the change.